doc/openapi.yaml

```diff
@@ -0,0 +1,42 @@
+swagger: "2.0"
```

Comment:
a lot of the tooling i've seen outside seems to prefer the .json format,  especially as it has the advatange to be able to be validated directly with the json-schema